openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1463,7 +1465,15 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setCertRequestedCallback)(TCN_STDARGS, jlon
         SSL_CTX_set_client_cert_cb(c->ctx, NULL);
     } else {
         jclass callback_class = (*e)->GetObjectClass(e, callback);
-        jmethodID method = (*e)->GetMethodID(e, callback_class, "requested", "(J[B[[B)Lio/netty/internal/tcnative/CertificateRequestedCallback$KeyMaterial;");
+
+        char* tmpSignature = netty_internal_tcnative_util_prepend(pPrefix, "io/netty/internal/tcnative/CertificateRequestedCallback$KeyMaterial;");
+        char* requestedSignature = netty_internal_tcnative_util_prepend("(J[B[[B)L", tmpSignature);
```

Comment:
consider just building this signature in the jni_onload, holding a reference to it, and freeing it in the jni_unload instead of building it on demand.